crates/libp2p-networking/src/network/behaviours/dht/mod.rs

```diff
@@ -323,6 +329,7 @@ impl DHTBehaviour {
                     .find(|(_, v)| *v >= NUM_REPLICATED_TO_TRUST)
                 {
                     // insert into cache
+                    // TODO we should not block here, this is called in poll.
```

Comment:
Good idea! will update.  I think the long term thing we should do is look into what kad already does for get caching.  I briefly looked into this last night and it seems like we check our storage first thing on a get query, which is more or less the same as this cache.  What I hadn't figured out yet is if we store every record we query for internally, i'll investigate more

Reply:
Hmm we can't spawn here because we can't hold onto `&self` to update the cache in the spawned task.  I don't see a way around this, so I think i'll leave as is for now and investigate the fix next

Reply:
Good point, probably we can clone

Reply:
clone seems to work